### ISSUE TITLE
prefer-tag-over-role false positives on React components

### ISSUE THREAD

**Issue by mihkeleidast** [jsx-eslint/eslint-plugin-jsx-a11y#1057]
For example:

```
<Image role="img" />
```

From [the docs](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/prefer-tag-over-role.md?plain=1), it seems like it's supposed to only target HTML elements, but currently it also reports failures on React components.

Just because a component has a `role` prop with values that happen to match one of the ARIA roles, it is not enough to suggest replacing such a component with a native HTML element. There is no way of knowing what the component does with the prop internally, or if it even makes it to a HTML element role attribute.
